### PR TITLE
EVG-14327 better handle errors in commit queue middleware

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: rest/route/middleware.go [hunk 1]
```diff
@@ -432,8 +432,8 @@ func (m *CommitQueueItemOwnerMiddleware) ServeHTTP(rw http.ResponseWriter, r *ht
 		pr, err := m.sc.GetGitHubPR(ctx, projRef.Owner, projRef.Repo, itemInt)
 		if err != nil {
 			gimlet.WriteResponse(rw, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
-				StatusCode: http.StatusInternalServerError,
-				Message:    "can't get information about PR",
+				StatusCode: http.StatusBadRequest,
+				Message:    fmt.Sprintf("unable to get pull request info, PR number may be invalid: %s", err),
```
Comment (ablack12) [inline rest/route/middleware.go:436]: Would it be helpful to return the PR number in this case?

**Review by ablack12 — APPROVED** [review]

File: rest/route/middleware.go (hunk 1 above)
Comment (john-m-liu) [reply rest/route/middleware.go]: yep done

**Comment by john-m-liu** [comment]
evergreen merge

**Comment by john-m-liu** [comment]
evergreen merge

**Pull request merged**